libs/api/src/lib/context.ts

```diff
@@ -1003,9 +1004,12 @@ export async function contextFromRequest(
       saveCustomer,
       failureURL,
       successURL,
-      user
+      user,
+      migrateToTargetPMid
```

Comment:
Why not accept here the subscription with memberpaln insted of some id? Like this we can determine the correct payment method from the subscription.

Reply:
The idea is, that we can tell this method, whether to migrate or not. Would like to keep it that way.